nifi-nar-bundles/nifi-alluxio-bundle/nifi-alluxio-processors/src/main/java/org/apache/nifi/processors/alluxio/GetAlluxio.java

```diff
@@ -0,0 +1,184 @@
+            .autoTerminateDefault(true)
+            .build();
+
+    private final static Set<Relationship> relationships;
+
+
+    /*
+     * Will ensure that the list of property descriptors is build only once.
+     * Will also create a Set of relationships
+     */
+    static {
+        List<PropertyDescriptor> _propertyDescriptors = new ArrayList<>();
+        _propertyDescriptors.addAll(descriptors);
+        _propertyDescriptors.add(READ_TYPE);
+        propertyDescriptors = Collections.unmodifiableList(_propertyDescriptors);
+
+        Set<Relationship> _relationships = new HashSet<>();
+        _relationships.add(REL_SUCCESS);
+        _relationships.add(REL_FAILURE);
+        _relationships.add(REL_SUCCESS_REQ);
+        relationships = Collections.unmodifiableSet(_relationships);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        createFileSystem(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile request = null;
+        if (context.hasIncomingConnection()) {
+            request = session.get();
+
+            // If we have no FlowFile, and all incoming connections are self-loops then we can continue on.
+            // However, if we have no FlowFile and we have connections coming from other Processors, then
+            // we know that we should run only if we have a FlowFile.
+            if (request == null && context.hasNonLoopConnection()) {
+                return;
+            }
+        }
+
+        final StopWatch stopWatch = new StopWatch(true);
+        final String uri = context.getProperty(URI).evaluateAttributeExpressions(request).getValue();
+        final AlluxioURI path = new AlluxioURI(uri);
+        final OpenFileOptions options = OpenFileOptions.defaults().setReadType(ReadType.valueOf(context.getProperty(READ_TYPE).getValue()));
+
+        FileInStream in = null;
+        FlowFile flowFile = null;
+
+        if(request == null) {
+            flowFile = session.create(request);
+        } else {
+            flowFile = session.create();
+        }
+
+        try {
+            final URIStatus status = fileSystem.get().getStatus(path);
+            flowFile = updateFlowFile(status, flowFile, session);
+
+            in = fileSystem.get().openFile(path, options);
+            final FileInStream toCopy = in;
+
+            flowFile = session.write(flowFile, new OutputStreamCallback() {
+                @Override
+                public void process(final OutputStream out) throws IOException {
+                    IOUtils.copy(toCopy, out);
```

Comment:
This is throwing errors in some cases and not bubbling out to the wrapping catch so there is no indication of issues.  One particular instance was:

> 016-06-19 01:12:52,512 ERROR [Timer-Driven Process Thread-8] o.a.nifi.processors.alluxio.PutAlluxio PutAlluxio[id=37744f8c-57fe-48c9-ac0b-e3fd255efc4b] An error occurred while writing file /nifi-630c823e-7492-4539-bf14-632dfa4b85e2; transferring to 'failure': java.lang.RuntimeException: No available Alluxio worker found
2016-06-19 01:12:52,513 ERROR [Timer-Driven Process Thread-8] o.a.nifi.processors.alluxio.PutAlluxio
java.lang.RuntimeException: No available Alluxio worker found
	at alluxio.client.block.AlluxioBlockStore.getOutStream(AlluxioBlockStore.java:174) ~[alluxio-core-client-1.0.1.jar:na]
	at alluxio.client.file.FileOutStream.getNextBlock(FileOutStream.java:266) ~[alluxio-core-client-1.0.1.jar:na]
	at alluxio.client.file.FileOutStream.write(FileOutStream.java:231) ~[alluxio-core-client-1.0.1.jar:na]
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1793) ~[commons-io-2.4.jar:2.4]
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1769) ~[commons-io-2.4.jar:2.4]
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:1744) ~[commons-io-2.4.jar:2.4]